walk/walk.go

```diff
@@ -35,12 +34,59 @@ type File struct {
 	RelPath string
 	Info    fs.FileInfo
 
-	// Cache is the latest entry found for this file, if one exists.
-	Cache *cache.Entry
+	// FormattedInfo is the result of os.stat after formatting the file.
+	FormattedInfo fs.FileInfo
+
+	// FormattersSignature represents the sequence of formatters and their config that was applied to this file.
+	FormattersSignature []byte
+
+	// CachedFormatSignature is the last FormatSignature generated for this file, retrieved from the cache.
+	CachedFormatSignature []byte
```

Comment:
If there was no entry in the cache this will be `nil`.

`nil` is analogous to an empty `[]byte` for operations like `len(CacheFormatSignature` or more importantly:

```go
// compare signature with last cache entry
if bytes.Equal(signature, file.CachedFormatSignature) {
	// If the signature is the same as the last cache entry, there is nothing to do.
	// We know from the hash signature that we have already applied this sequence of formatters (and their config) to
	// this file.
	// When we applied the formatters, the file had the same mod time and file size.
	return false, nil
}
```